test/language/statements/variable/async-async-identifier-assignment-parse.js

```diff
@@ -0,0 +1,31 @@
+// This file was procedurally generated from the following sources:
+// - src/async-async/identifier-assignment-parse.case
+// - src/async-async/parse/var.template
+/*---
+description: async is not a valid syntax as an identifier reference with a default value (Object Binding Pattern in a variable statement)
+esid: prod-ObjectBindingPattern
+flags: [generated]
+info: |
+    VariableStatement:
+    var VariableDeclarationList;
+
+    VariableDeclarationList:
+    VariableDeclaration
+    ...
+
+    VariableDeclaration:
+    ...
+    BindingPattern Initializer
+
+    BindingPattern:
+      ObjectBindingPattern
+      ...
+
+    ObjectBindingPattern:
+      {}
+      { BindingPropertyList }
+      { BindingPropertyList, }
+---*/
+
+
+assert.throws.early(SyntaxError, `var {async async = 0} = {};`);
```

Comment:
this is now generated code using a similar model for the early parse error.

If the pre-parse and parse are different, this approach must be reasonable, even if longer.

I tried to reuse the templates instead of duplicating each case file. For that I would need to define the negative flags there instead of each case. The test generation tool does not support that yet.